src/Microsoft.Tye.Core/ValidateIngressStep.cs

```diff
@@ -140,29 +140,28 @@ public override async Task ExecuteAsync(OutputContext output, ApplicationBuilder
             else
             {
                 // If we get here then we should deploy the ingress controller.
+
+                // For some reason, the first time we apply the ingress controller, an exception is thrown
```

Comment:
It's not really *for some reason* - this is plot hole in how CRDs + validating webhooks work. Basically if you create a new CRD + a new webhook to validate it - what should the behavior be before the webhook starts? The author of the webhook can choose to be strict, and say no one gets to create one of these without my approval. There's no policy for "wait 30 sec to see if the webhook starts, and try again".

That's the problem you're likely seeing. Unfortunately helm doesn't solve it either :-/

Reply:
Yeah that's the error I'm seeing.